### PR TITLE
Add extras features docs to new model checklist

### PR DESCRIPTION
**Pull request by gsnider2195**
Housekeeping

image

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
Great stuff!

Isn't `graphql` an extras_feature as well? Looks like that one got missed.

Would it make sense to add a recommendation for a "default/minimal" set of extras_features, probably `custom_links, custom_validators, export_templates, graphql, webhooks` for all user-facing models?

File: nautobot/docs/development/core/model-checklist.md [hunk 1]
```diff
@@ -34,6 +34,22 @@ article ul li {
     - Update ConfigContextQuerySet, ConfigContextFilterSet, ConfigContext forms, edit and detail views and HTML templates
 - _optional_ Expose any new relevant Python APIs in `nautobot.apps` namespace for App consumption
 
+#### Extras Features
```
Comment (glennmatthews) [inline nautobot/docs/development/core/model-checklist.md:37]: Probably the mention of `extras_features` at line 23 above should link to this heading.

**Comment by gsnider2195** [comment]
> Great stuff!
> 
> Isn't `graphql` an extras_feature as well? Looks like that one got missed.
> 
> Would it make sense to add a recommendation for a "default/minimal" set of extras_features, probably `custom_links, custom_validators, export_templates, graphql, webhooks` for all user-facing models?

Not sure how I missed graphql. Those are good suggestions. I'll make those changes.

**Review by glennmatthews — APPROVED** [review]

File: nautobot/docs/development/core/model-checklist.md [hunk 2]
```diff
@@ -34,6 +34,25 @@ article ul li {
...
+- `custom_links`: Models that can display `CustomLinks` on the object's detail view (by default, all models that use `generic/object_retrieve.html` as a base template support custom links)
+- `custom_validators`: Models that can support custom `clean` logic by implementing a [`CustomValidator`](../apps/api/platform-features/custom-validators.md)
+- `dynamic_groups`: Models that can be assigned to a `DynamicGroup`, used for limiting the choices for the `DynamicGroup.content_type` form field
+- `export_template_owners`: Models that can be assigned to the `owner` GenericForeignKey field on an `ExportTemplate`
+- `export_templates`: Models that can be exported using an `ExportTemplate`, used for limiting the choices for the `ExportTemplate.content_type` field
+- `graphql`: Models that should be exposed through the GraphQL API, used to build the list of registered models to build the GraphQL schema
```
Comment (gsnider2195) [inline nautobot/docs/development/core/model-checklist.md:47]:
```suggestion
- `graphql`: Models that should be exposed through the [GraphQL API](../apps/api/models/graphql.md), used to build the list of registered models to build the GraphQL schema
```

**Pull request merged**